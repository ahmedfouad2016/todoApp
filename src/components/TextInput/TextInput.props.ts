import {StyleProp, TextInputProps, TextStyle} from 'react-native';

export interface InputType extends TextInputProps {
  onChangeText: (text: string) => void;
  value?: string;
  placeholder?: string;
  inputStyle?: StyleProp<TextStyle>;
  forwardedRef?: any;
}
