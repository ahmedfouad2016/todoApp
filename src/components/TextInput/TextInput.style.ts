import {I18nManager, StyleSheet} from 'react-native';

export default StyleSheet.create({
  input: {
    height: 40,
    borderWidth: 1,
    padding: 10,
    textAlign: I18nManager.isRTL ? 'right' : 'left',
  },
});
