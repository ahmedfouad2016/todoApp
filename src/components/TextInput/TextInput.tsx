import React from 'react';
import {TextInput as RNTextInput, StyleSheet} from 'react-native';
import {InputType} from './TextInput.props';

import styles from './TextInput.style';

export const TextInput = (props: InputType) => {
  const {value, onChangeText, placeholder, forwardedRef, inputStyle, ...rest} =
    props;
  const style = StyleSheet.flatten([styles.input, inputStyle]);
  return (
    <RNTextInput
      {...rest}
      testID="text-input"
      style={style}
      onChangeText={onChangeText}
      value={value}
      placeholder={placeholder}
      ref={forwardedRef}
    />
  );
};
