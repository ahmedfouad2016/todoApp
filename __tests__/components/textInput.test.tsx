import 'react-native';
import React from 'react';
import {TextInput} from '@components/TextInput';

import {fireEvent, render} from '@testing-library/react-native';

describe('TextInput Component', () => {
  it('should TextInput render ', () => {
    const wrapper = render(<TextInput onChangeText={() => null} />);
    expect(wrapper.toJSON()).toMatchSnapshot();
  });

  it('should TextInput display placeholder', () => {
    const placeholder = 'add list name';
    const {queryByPlaceholderText} = render(
      <TextInput onChangeText={() => null} placeholder={placeholder} />,
    );
    const comp = queryByPlaceholderText(placeholder);
    expect(comp).not.toBeNull();
  });

  it('should TextInput display default value', () => {
    const value = 'my list';
    const {queryByDisplayValue} = render(
      <TextInput onChangeText={() => null} value={value} />,
    );
    const comp = queryByDisplayValue(value);
    expect(comp).not.toBeNull();
  });

  it('should TextInput inputStyle reflect text input style', () => {
    const result = {height: 300, margin: 25};
    const {getByTestId} = render(
      <TextInput onChangeText={() => null} inputStyle={result} />,
    );
    const comp = getByTestId('text-input');
    expect(comp.props.style).toMatchObject(result);
  });

  it('should fire onChangeText event', () => {
    const onChangeFn = jest.fn();
    const param = 'my list';
    const {getByTestId} = render(<TextInput onChangeText={onChangeFn} />);
    fireEvent(getByTestId('text-input'), 'onChangeText', param);
    expect(onChangeFn).toHaveBeenCalled();
    expect(onChangeFn).toHaveBeenCalledWith(param);
  });
});
