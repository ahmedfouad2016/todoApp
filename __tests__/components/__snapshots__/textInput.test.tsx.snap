// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TextInput Component should TextInput render  1`] = `
<TextInput
  onChangeText={[Function]}
  style={
    Object {
      "borderWidth": 1,
      "height": 40,
      "padding": 10,
    }
  }
  testID="text-input"
/>
`;
